### PR TITLE
:bug: Source MySQL: transform binary data base64 format

### LINKED ISSUE #5878: MySql source : source binary node transforms to text node
Found during work on #3932

## Current Behavior
Initial investigation showed that the data type test gets airbyte messages with binary values as text.
As a result, check `isBinary` returns false and the value contains useless data (not even string version).

Note! The issue is not investigated fully. There might be a global issue of binary values. 

## Expected Behavior
Binary values in Airbyte message data should be in Binnary nodes for proper processing.

### PR DESCRIPTION
**Pull request by mkhokh-33**
## What
*MySql source handle binary data as base64 encoded string, fixed integrations data type tests* 
*Fixed cdc handling of binary data as base64 encoded string and cdc integrations data type*
*Provide additional properties in JSON schema for binary data*
*Added tests to check how BinaryNode will be serialized*
## How
*Configured MySqlCdcProperties* with binary.handling.mode = base64 to make sure that binary data represented as a base64-encoded String

## Recommended reading order
1. `x.java`
2. `y.json`
3. `y.md`

## 🚨 User Impact 🚨
For AirbyteStream in jsonShema will appear new additional property contentEncoding:
jsonSchema={"type":"object","properties":{"name":{"type":"string","contentEncoding":"base64"},"id":{"type":"string"}}}
as far as I see - https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py#L483
it should not affect how binary data will be represented in destination with normalization

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated 
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [x] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-protocol/models/src/main/java/io/airbyte/protocol/models/JsonSchemaPrimitive.java [hunk 1]
```diff
@@ -4,11 +4,26 @@
...
+
 public enum JsonSchemaPrimitive {
-  STRING,
-  NUMBER,
-  OBJECT,
-  ARRAY,
-  BOOLEAN,
-  NULL;
+
+  STRING_BINARY(ImmutableMap.of("type", "string", "contentEncoding", "base64")),
```
Comment (VitaliiMaltsev) [inline airbyte-protocol/models/src/main/java/io/airbyte/protocol/models/JsonSchemaPrimitive.java:11]: @tuliren can we use JsonSchemaPrimitive as is in scope of this PR or better wait until https://github.com/airbytehq/airbyte/issues/7944 will be implemented?

Comment (tuliren) [reply airbyte-protocol/models/src/main/java/io/airbyte/protocol/models/JsonSchemaPrimitive.java:11]: Yes, I think the current implementation looks good.

**Review by tuliren — APPROVED** [review]

**Review by VitaliiMaltsev — APPROVED** [review]

**Review by DoNotPanicUA — APPROVED** [review]

**Review by sashaNeshcheret — APPROVED** [review]

**Pull request merged**